File: JAVA/java/src/SpecialArrayWithXElementsGreaterThanorEqualX.java
public class SpecialArrayWithXElementsGreaterThanorEqualX {
        public int specialArray(int[] nums) {
            int N = nums.length;
            int[] freq = new int[N + 1];

            for (int i = 0; i < N; i++) {
                freq[Math.min(N, nums[i])]++;
            }

            int numGreaterThanOrEqual = 0;
            for (int i = N; i >= 1; i--) {
                numGreaterThanOrEqual += freq[i];
                if (i == numGreaterThanOrEqual) {
                    return i;
                }
            }

            return -1;
        }

        public static void main(String[] args) {
            SpecialArrayWithXElementsGreaterThanorEqualX ob=new SpecialArrayWithXElementsGreaterThanorEqualX();
            System.out.print(ob.specialArray(new int[] {2,5}));

         }

}

File: python/Special Array With X Elements Greater Than or Equal X.py
def specialArray(nums):
    N = len(nums)

    freq = [0] * (N + 1)
    for num in nums:
        freq[min(N, num)] += 1

    num_greater_than_or_equal = 0
    for i in range(N, 0, -1):
        num_greater_than_or_equal += freq[i]
        if i == num_greater_than_or_equal:
            return i

    return -1
print(specialArray([2,5]))